components/transfer/commons/transfer_row.lua

```diff
@@ -0,0 +1,324 @@
+local Class = require('Module:Class')
+local Json = require('Module:Json')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Namespace = require('Module:Namespace')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+local Variables = require('Module:Variables')
+
+local PlayerExt = Lua.import('Module:Player/Ext/Custom')
+local PositionConvert = Lua.requireIfExists('Module:PositionName/data', {loadData = true})
+local TransferRowDisplay = Lua.import('Module:TransferRow/Display')
+local References = Lua.import('Module:Transfer/Refences')
+
+local HAS_PLATFORM_ICONS = Lua.moduleExists('Module:Platform/data')
+local VALID_CONFIDENCES = {
+	'certain',
+	'likely',
+	'possible',
+	'unlikely',
+	'unknown',
+}
+
+---@class TransferRow: BaseClass
+---@field config {storage: boolean, isRumour: boolean}
+---@field transfers transfer[]
+---@field args table
+---@field baseData table
+local TransferRow = Class.new(
+	---@param args table
+	---@return self
+	function(self, args)
+		self.args = args
+
+		return self
+	end
+)
+
+---@return self
+function TransferRow:read()
+	self.config = self:readConfig()
+	self.transfers = self:readInput()
+
+	return self
+end
+
+---@return {storage: boolean, isRumour: boolean}
+function TransferRow:readConfig()
+	local isRumour = Logic.readBool(self.args.isRumour)
+	return {
+		storage = not isRumour and
+			not Logic.readBool(self.args.disable_storage) and
+			not Logic.readBool(Variables.varDefault('disable_LPDB_storage'))
+			and Namespace.isMain(),
+		isRumour = isRumour,
+	}
+end
+
+---@return transfer[]
+function TransferRow:readInput()
+	local transferSortIndex = tonumber(Variables.varDefault('transfer_sort_index')) or 0
+
+	self.baseData = self:_readBaseData()
+	local players = self:readPlayers()
+	self.references = self:_readReferences(#players)
+
+	local transfers = Array.map(players, function(player, playerIndex)
+		local transfer = self:_convertToTransferStructure{
+			player = player,
+			index = playerIndex,
+			sortIndex = transferSortIndex
+		}
+
+		transferSortIndex = transferSortIndex + 1
+		return transfer
+	end)
+
+	Variables.varDefine('transfer_sort_index', transferSortIndex)
+
+	return transfers
+end
+
+---@return table
+function TransferRow:_readBaseData()
+	local args = self.args
+
+	---@param str string?
+	---@return string?
+	local ucFirst = function(str)
+		return str and String.upperCaseFirst(str) or nil
+	end
+
+	---@param data string[]
+	---@return string[]
+	local switchInactiveIfAppropriate = function(data)
+		if data[1] ~= 'Inactive' or Logic.isEmpty(data[2]) or data[2] == 'Inactive' then
+			return data
+		end
+		return {data[2], data[1]}
+	end
+
+	local fromRole = switchInactiveIfAppropriate(Array.map({args.role1, args.role1_2}, ucFirst))
+	local toRole = switchInactiveIfAppropriate(Array.map({args.role2, args.role2_2}, ucFirst))
+
+	local date = args.date_est or args.date
+	local fromDate = TransferRow._shiftDate(date)
```

Comment:
I guess this is done to account for rebrand transfers?
IMO it should be behind a flag and/or only active if team1 == team2, as it might just happen that a HTT switch happens and the next day there's a transfer, which would have the wrong template then.

Reply:
the old module does it already
so changing it would be a functional change that could cause issues
https://liquipedia.net/commons/index.php?title=Module:Transfer&action=edit#mw-ce-l433

Reply:
we will see, would like to keep it oos for now though^^